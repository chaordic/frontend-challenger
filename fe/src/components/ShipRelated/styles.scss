@import '../../global-styles/settings.global.scss';

.ship-related {
    padding: 5px;
    color: var(--green);
    text-decoration: none;
    width: 30px;
    text-align: center;
    height: 20px;
    color: var(--green);
    border: 1px solid var(--green);
    border-radius: 3px;
    cursor: pointer;
    float: left;
    margin-right: 5px;
}