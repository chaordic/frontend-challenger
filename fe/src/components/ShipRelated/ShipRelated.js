import React from 'react';
import { string, func } from 'prop-types';
import cssModules from 'react-css-modules';
import styles from './styles.scss';

const ShipRelated = ({ id, onClick }) => (
  <a
    className={styles['ship-related']}
    key={id}
    href={`#shipItem-${id}`}
    onClick={onClick}
  >
    {id}
  </a>
);

ShipRelated.propTypes = {
  id: string.isRequired,
  onClick: func,
};

ShipRelated.defaultProps = {
  onClick: () => {},
};

export default cssModules(ShipRelated, styles, { allowMultiple: true });
