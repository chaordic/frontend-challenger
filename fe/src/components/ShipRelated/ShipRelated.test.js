import React from 'react';
import renderer from 'react-test-renderer';
import ShipRelated from './ShipRelated';

const props = {
  onClick: jest.fn(),
  id: 'F1',
};

describe('(Component) ShipRelated', () => {
  it('should render an ShipRelated with result', () => {
    const tree = renderer.create(<ShipRelated {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
