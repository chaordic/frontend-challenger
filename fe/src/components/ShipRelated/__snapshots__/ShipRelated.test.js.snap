// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ShipRelated should render an ShipRelated with result 1`] = `
<a
  href="#shipItem-F1"
  onClick={[MockFunction]}
>
  F1
</a>
`;
